### ISSUE TITLE
Using barcode codeboxes and code128 support

### ISSUE THREAD

**Issue by garidan** [sdaps/sdaps#215]
Hi, SDAPS 1.9.6 on standard Ubuntu installed by apt-get: I cannot get code128 barcode codebox recognized into a text field.
I followed instructions at https://github.com/sdaps/sdaps/issues/2 and https://sdaps.org/class-doc/customlayout.html, but I use code128 barcodes but I understand barcode standard isn't to be declared.
Codebox is displayed and annotations and GUI tell me the block is recognized but no text is filled. I thought it was a problem with barcode quality, but now I prepare a 600 dpi tiff scan and convert to B/W before adding, quality is perfect.
What's wrong and how do I debug this: is there a place to look for isolated image regions given to barcode library ?
Is there something new in 1.9.9  about codeboxes ? 
Thanks

**Comment by benzea** [comment]
Yeah, the problem is that the code currently assumes it is a QR code.

You can work around it by modifying `sdaps/recognize/buddies.py`. In the `Codebox` class there is a call to `read_barcode` with an argument of `QRCODE`. Change that to `CODE128` and it will work.

**Comment by garidan** [comment]
I confirm it works on 1.9.6 with the patch. It's at the end of that file.
I note this other patch for GUI is usefull too for people using barcodes. A not recognized codebox would get a blank text field and let this other bug show up.
https://github.com/sdaps/sdaps/commit/6732330679556d10846eb0b62e352242073c4acd
